File: prime_number_primer/prime_number_primer__greatest_common_divisor.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        int[] a = new int[n];

        foreach( int i in Range( 0, n ) ) {
            a[i] = int.Parse( ReadLine() );
        }
        WriteLine( GcdM( a, n ) );
        return;
    }

    static int GcdM( int[] a, int n )
    {
        int o = a[0];

        foreach( int i in Range( 1, n - 1 ) ) {
            o = Gcd( o, a[i] );
        }
        return o;
    }

    static int Gcd( int a, int b )
    {
        return b == 0 ? a : Gcd( b, a % b );
    }
}

File: prime_number_primer/prime_number_primer__greatest_common_divisor.py
# 実行時間 10ms
import math
import sys

def gcdM(a, n):
    o = a[0]

    for i in range(1, n):
        o = math.gcd(o, a[i])
    return o

r = sys.stdin.readline
n = int(r())
a = [0] * n

for i in range(n):
    a[i] = int(r())
print(gcdM(a, n))
